pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
@deads2k IIUC, what we do is just:
1) read the auth information from config file or somewhere else;
2) Fill the `DockerConfigEntry` with the auth information;
3) Convert the `map[string]DockerConfigEntry` into `map[string]dockertypes.AuthConfig`
4) Put the `AuthConfig` into `DockerKeyring`

When we pull image, we just:
5) Lookup the `AuthConfig` from `DockerKeyring`
6) Encode the `AuthConfig` in base64
7) Pull image with it.

If this is the workflow, I think it would be better to just generate the `AuthConfig` at 2)